velero-plugins/build/restore.go

```diff
@@ -47,6 +48,39 @@ func (p *RestorePlugin) Execute(input *velero.RestoreItemActionExecuteInput) (*v
 	p.Log.Info(fmt.Sprintf("Found new dockercfg secret: %v", secret))
 	build = createNewPushSecret(build, secret)
 
+	// Swap out stale imageRefs
+	// TEMP: Getting registry information dynamically because annotations do not exist
+	version, err := common.GetServerVersion()
+	if err != nil {
+		p.Log.Error("Error getting server version: ", err)
+		return nil, err
+	}
+	registry, err := common.GetRegistryInfo(version.Major, version.Minor)
+	if err != nil {
+		p.Log.Error("Error getting registry information: ", err)
+		return nil, err
+	}
+	name := build.Spec.Strategy.SourceStrategy.From.Name
+	shaSplit := strings.Split(name, "@")
+	if len(shaSplit) < 2 {
+		err = fmt.Errorf("unexpected image reference [%v]", name)
+		return nil, err
+	}
+	sha := shaSplit[1]
+	splitName := strings.Split(shaSplit[0], "/")
+	if len(splitName) < 2 {
+		err = fmt.Errorf("unexpected image reference [%v]", name)
+		return nil, err
+	}
+	namespacedName := splitName[len(splitName)-2:]
+	newName := fmt.Sprintf("%s/%s/%s@%s", registry, namespacedName[0], namespacedName[1], sha)
+
+	// Replace all imageRefs
+	build.Spec.Strategy.SourceStrategy.From.Name = newName
+	for _, trigger := range build.Spec.TriggeredBy {
+		trigger.ImageChangeBuild.FromRef.Name = newName
```

Comment:
Do we need to confirm that the name we're replacing is an exact match for the old name from above (`name`)?